addons/marketing_card/models/card_campaign.py

```diff
@@ -0,0 +1,288 @@
+            default_vals.setdefault('card_element_ids', [
+                Command.create({'card_element_role': 'background', 'render_type': 'image'}),
+                Command.create({'card_element_role': 'header', 'render_type': 'text'}),
+                Command.create({'card_element_role': 'subheader', 'render_type': 'text'}),
+                Command.create({'card_element_role': 'section_1', 'render_type': 'text'}),
+                Command.create({'card_element_role': 'subsection_1', 'render_type': 'text'}),
+                Command.create({'card_element_role': 'subsection_2', 'render_type': 'text'}),
+                Command.create({'card_element_role': 'button', 'render_type': 'text'}),
+                Command.create({'card_element_role': 'image_1', 'render_type': 'image'}),
+                Command.create({'card_element_role': 'image_2', 'render_type': 'image'}),
+            ])
+        return default_vals
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    body_html = fields.Html(related='card_template_id.body', render_engine="qweb")
+
+    card_count = fields.Integer(compute='_compute_card_stats')
+    card_click_count = fields.Integer(compute='_compute_card_stats')
+    card_share_count = fields.Integer(compute='_compute_card_stats')
+
+    card_template_id = fields.Many2one('card.template', string="Design", default=_default_card_template_id, required=True)
+    card_element_ids = fields.One2many('card.campaign.element', inverse_name='campaign_id', copy=True)
+    image_preview = fields.Image(compute='_compute_image_preview', readonly=True, store=True, attachment=False)
+    link_tracker_id = fields.Many2one('link.tracker', ondelete="restrict")
+    res_model = fields.Selection(string="Model Name", selection=[('res.partner', 'Contact')], compute="_compute_res_model",
+                                 copy=True, precompute=True, readonly=False, required=True, store=True)
+
+    post_suggestion = fields.Text(help="Default text when sharing on X")
+    preview_record_ref = fields.Reference(string="Preview Record", selection="_selection_preview_record_ref")
+    preview_record_url = fields.Char('Preview Record Link', compute="_compute_preview_record_url")
+    reward_message = fields.Html(string='Thanks to You Message')
+    reward_target_url = fields.Char(string='Reward Link')
+    tag_ids = fields.Many2many('card.campaign.tag', string='Tags')
+    target_url = fields.Char(string='Shared Link')
+    target_url_click_count = fields.Integer(related="link_tracker_id.count")
+
+    user_id = fields.Many2one('res.users', string='Responsible', default=lambda self: self.env.user, domain="[('share', '=', False)]")
+    utm_campaign_id = fields.Many2one('utm.campaign', string='UTM Campaign', ondelete='set null')
+
+    random_token = fields.Char(readonly=True, required=True, default=lambda self: secrets.token_urlsafe())
```

Comment:
Not sure we need it, we could just use the tool to create a hash based on secret and it should be sufficient, isn't it ?